src/app/shared/owner-guard.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import {
+  CanActivate,
+  CanActivateChild,
+  Router,
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot
+} from '@angular/router';
+
+import { Observable, Subscription } from 'rxjs';
+
+import { AuthenticationService, UserService, User } from 'ngx-login-client';
+
+import { LoginService } from './login.service';
+
+/**
+ * A guard which checks that the owner of the resource is the same as the one
+ * who is logged in. This will also fail if the user is not logged in, making
+ * it a superset of the AuthGuard service.
+ */
+@Injectable()
+export class OwnerGuard implements OnDestroy {
+  private subscription: Subscription;
+
+  constructor(
+    private authService: AuthenticationService,
+    private router: Router,
+    private loginService: LoginService,
+    private userService: UserService
+  ) { }
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+    if (this.authService.isLoggedIn()) {
+      return Observable.create(observer => {
+        this.subscription = this.userService.loggedInUser.subscribe(user => {
+          observer.next(this.userMatchesOwnerFromRoute(user));
```

Comment:
I added a helper function to the context service that checks if a user is viewing his own stuff (see viewingOwnContext() in shared/context.service.ts).  It will simplify this code to use that.